enterprise/backend/src/metabase_enterprise/uploads/api.clj

```diff
@@ -0,0 +1,37 @@
+(ns metabase-enterprise.uploads.api
+  (:require
+   [compojure.core :refer [DELETE]]
+   [metabase.api.common :as api]
+   [metabase.api.routes.common :refer [+auth]]
+   [metabase.models.interface :as mi]
+   [metabase.upload :as upload]
+   [metabase.util.i18n :refer [tru]]
+   [metabase.util.malli.schema :as ms]
+   [toucan2.core :as t2]))
+
+(api/defendpoint GET "/"
+  "Get all `Tables` visible to the current user which were created by uploading a file."
+  []
+  (as-> (t2/select :model/Table, :active true, :is_upload true, {:order-by [[:name :asc]]}) tables
+        ;; See https://github.com/metabase/metabase/issues/41023
+        (map #(update % :schema str) tables)
```

Comment:
It's interesting that we need this. We could do this in the after-select hook tho

Reply:
If I recall correctly, that option was considered when Kamil and Braden explored all the approaches, but we decided this was a convention only for the frontend and we wanted to keep using nil's within the backend logic.

I pointed out the ability to extend `to-json` to `:model/Table` to do things generically for the API (assuming nothing strips the toucan instance wrapper) but in the end I think we decided to keep things simple as its exposed in so few places.